### PR TITLE
Split the Climatic > Tidy and Examine menu; and added Pivot Table to the Climatic > Tidy and Explore menu

### LINKED ISSUE #8346: Add Pivot Table to the Climatic > Tidy and Explore menu
Here is the dialog:

Please could it be added to the bottom of the Climatic > Tidy Menu?

If the data frame is not defined as climatic, then it is empty, just as in the DEscribe menu.
In addition, when the data are defined as climatic could it (by default) be completed as shown below:

image

This assumes there are all 3 of year, month and day of month defined (they usually are)  some are not defined, then just leave them out.  If rain is not defined, but another climatic element is defined, then put it in.  (If no climatic elements are defined - that's very rare) it still works with the numeric variable still ticked, but the control left empty..  It gives a blank table and that's ok. Use Average and Heat Map as shown above.

@N-thony who could do this?  Is it another task for @Fidel365 ?

### LINKED ISSUE #8348: Split the Climatic > Tidy and Examine menu
Please proceed with these changes.  (I earlier wrote:  Please do not implement this issue, until others, including some of @N-thony and @Patowhiz and @rachelkg and @lloyddewit have given their views.)  I have some views now, and would like to proceed!

The climatic > Tidy and Examine menu is currently as follows:

image

I have proposed the addition of Pivot Tables to the bottom of this menu.  This will make it the longest menu un the whole of R-Instat! (The Prepare > Data Frame is the only other one that comes close.)

I have recently been pretty happy with some of the menu changes - particularly to the describe menu.

I have also been happy with the way that the climatic courses go steadily down the climatic menu.  I now tentatively propose we split this climatic Tidy and Examine menu.  

If so the first half becomes `Tidy Data` and the Second would be `Examine and Edit. `

Then the Tidy Data is the first 2 sections, so down to Append.

The Examine and Edit would swap the order of the remaining parts, so the 7 items would become:

One Variable Summarise...
One Variable Graph...
One Variable Frequencies...
----------------------
Pivot Table...
--------------------
Setup for Data Editing
Daily Data Editing
Compare Columns

(Note the change in name - no entry anymore that's for Climsoft!)

I am growing to like this as I write it.  

In addition I find I sometimes need the `Transform Text` and `Split Text` dialogs from the Prepare menu.  With the new setup I propose they could become a small 3rd section of the `Climatic > Tidy Data` menu.

What do you think?

### PR DESCRIPTION
**Pull request by MeSophie**
Fixes #8348
Fixes #8346
@rdstern and @N-thony this new PR is because the PR #8377 has conflict. 
For issue #8346 we are still investigate on how to autofill multiple receiver.
The name of `Tidy Data` is still `Tidy and Examine` because the text need to be change in the data base. So @lloyddewit please can you do it for me oe let me know when you merge the PR that has data base modification.

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
@MeSophie this is coming on well.  
a) Could you start with the Numeri9c Variable checked.
b) In the Climatic > Check Data > Inventory the multiple receiver is filled.  I wonder if the code there might help.  Of course, that is with climatic variables and you want others in this dialog.

**Comment by N-thony** [comment]
> @MeSophie this is coming on well. a) Could you start with the Numeri9c Variable checked. b) In the Climatic > Check Data > Inventory the multiple receiver is filled. I wonder if the code there might help. Of course, that is with climatic variables and you want others in this dialog.

@MeSophie I'm ready to help you with this.

**Comment by MeSophie** [comment]
@rdstern Please can you check if this is the change that you want? 
I went to check Inventory plot dialog. The multiple receiver is empty but the selector shows just numeric variables.

image

**Comment by rdstern** [comment]
@MeSophie I got it wrong in #8346 !  Just trying to make life more difficult for you - I hope you understand!

The columns should be the months and the day variable should be in the multiple receiver.

image

Sorry about that!

**Comment by MeSophie** [comment]
> 

@rdstern Please can you check the dialog again?

**Comment by rdstern** [comment]
@MeSophie that's better.  Can you start with the Numeric Variable checked please.

Then, ideally you would include the year and day in the multiple receiver as well.  But if that remains tricky let's merge with the progress so far.  That can be considered later.

**Comment by MeSophie** [comment]
> @MeSophie that's better. Can you start with the Numeric Variable checked please.
> 
> Then, ideally you would include the year and day in the multiple receiver as well. But if that remains tricky let's merge with the progress so far. That can be considered later.

@rdstern Please can you explain what do you means by Numeric Variables Checked?

**Comment by rdstern** [comment]
@MeSophie in the dialog there is a checkbox called `Numeric Variable (Optional).`  

When it is checked, then you can see the single receiver and you have correctly added rain there by default.  But you need to  add the tick to see it.  

I hope that's clear now?

**Comment by MeSophie** [comment]
> @MeSophie in the dialog there is a checkbox called `Numeric Variable (Optional).`
> 
> When it is checked, then you can see the single receiver and you have correctly added rain there by default. But you need to add the tick to see it.
> 
> I hope that's clear now?

Yes @rdstern  Thank you.

**Comment by MeSophie** [comment]
@rdstern please can you check once more?

**Comment by rdstern** [comment]
This seems to have become quite hard work for you.  Here is the dialog now:

image

The checkbox is correctly checked.  But I would like the focus to still be on the multiple receiver, even if it remains empty.

And we now seem to have lost the statistics for the summary.  Those controls were full, but they are not now.

**Comment by MeSophie** [comment]
@rdstern Please can you check the changes

**Comment by rdstern** [comment]
@MeSophie that looks much better.  I was ready to suggest merging and we leave the filling of the multiple receiver for later.  But when you get the pivot table it is slightly different:

Here is the dialog:

image

Looks fine, but here is the pivot table:

image


I assume, because the focus is back on the multiple receiver, it has not recognised the rain as the numeric variable?

**Comment by MeSophie** [comment]
@rdstern please can you check the new change?

**Review by rdstern — APPROVED** [review]
@MeSophie well done.  I am approving.  Please make an issue of the small remaining addition, but this is a big change and important to get into the next version.  @lloyddewit I hope it passes your tests soon?

File: instat/dlgThreeVariablePivotTable.vb [hunk 1]
```diff
@@ -299,18 +313,55 @@ Public Class dlgThreeVariablePivotTable
...
+                strDataFrame = ucrSelectorPivot.ucrAvailableDataFrames.cboAvailableDataFrames.Text
+                strMonthCol = frmMain.clsRLink.GetClimaticColumnOfType(strDataFrame, "month_label")
+                strRainCol = frmMain.clsRLink.GetClimaticColumnOfType(strDataFrame, "rain_label")
+                strYearCol = frmMain.clsRLink.GetClimaticColumnOfType(strDataFrame, "year_label")
+                strDayCol = frmMain.clsRLink.GetClimaticColumnOfType(strDataFrame, "day_label")
+                If strRainCol <> "" Then
+                    ucrReceiverAdditionalRowFactor.Add(strRainCol, strDataFrame)
+                End If
+
+                If strMonthCol <> "" Then
```
Comment (lloyddewit) [inline instat/dlgThreeVariablePivotTable.vb:344]: Safer to use `String.IsNullOrEmpty()` function?

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@MeSophie thank you for the clean implementation. I only had very small comments.

**Review by rdstern — COMMENTED** [review]
@MeSophie The dialog is now completed - seemingly correctly (except for the multiple receiver of course), but the display seems to have reverted to givning the counts and not the averages of Precip?

**Comment by MeSophie** [comment]
> The dialog is now completed - seemingly correctly (except for the multiple receiver of course), but the display seems to have reverted to givning the counts and not the averages of Precip?

@rdstern this is what I have from my side. For the multiple receiver problem I will check it tomorrow with Vitalis I try it alone and is still not working.
image

**Review by rdstern — COMMENTED** [review]
@MeSophie I think that is because you are being too conscientious when you are testing!
a) I tried with dodoma - as you are - and put the year and day myself into the multiple reciever.  And it worked perfectly.
b) Then I tried again, but forgetting to put the year and day into the receiver.  (You can always add them later, once in the pivot table.  And it now gave me the count.
c) Then I went back to the dialog and pressed Reset and then did it all perfectly again, and it now still gives me the Counts!

Can you try going back repeatedly and also not always being sensible with the multiple receiver - perhaps just putting the year, or the day in, etc.

**Comment by MeSophie** [comment]
@rdstern I couldn't solve the autofill multiple receiver problem. What Vitalis explain to me is that in inventory plot. The multiple receiver is fill automaticaly according to the variable define as `element`  in metadata. So we still need to investigate on how to autofill multiple variable with specific climatic parameters.

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**